Source/SwiftyDropbox/Shared/Handwritten/ChunkInputStream.swift

```diff
@@ -0,0 +1,118 @@
+///
+/// Copyright (c) 2020 Dropbox, Inc. All rights reserved.
+///
+/// Swift translation of the DBChunkInputStream ObjC class, without overrides of CFReadStream bridged methods
+/// which are not necessary now.
+/// Original files:
+/// https://github.com/dropbox/SwiftyDropbox/blob/6.0.1/Source/SwiftyDropbox/Shared/Handwritten/DBChunkInputStream.h
+/// https://github.com/dropbox/SwiftyDropbox/blob/6.0.1/Source/SwiftyDropbox/Shared/Handwritten/DBChunkInputStream.m
+///
+
+import Foundation
+
+/// Subclass of `InputStream` to enforce "bounds" on file stream, for chunk uploading.
+class ChunkInputStream: InputStream, StreamDelegate {
+    private let internalStream: InputStream?
+    private var internalStreamStatus = InputStream.Status.notOpen
+    private var startBytes: Int
+    private var totalBytesToRead: Int
+    private var totalBytesRead: Int
+    private weak var streamDelegate: StreamDelegate?
+
+    ///
+    /// Full constructor.
+    ///
+    /// - Parameters:
+    ///     - fileUrl: The file to stream.
+    ///     - startBytes: The starting position of the file stream, relative to the beginning of the file.
+    ///     - endBytes: The ending position of the file stream, relative to the beginning of the file.
+    /// - returns An initialized ChunkInputStream instance.
+    ///
+    init(fileUrl: URL, startBytes start: Int, endBytes end: Int) {
+        precondition(end > start, "End location \(end) needs to be greater than start location \(start)")
+
+        self.internalStream = InputStream(url: fileUrl)
```

Comment:
I agree with you, however I don't feel very comfortable making this change right now, because making this failable would require some extra changes to where this class is used which is the `FilesRoutes` extension in `CustomRoutes.swift`. The code there was kinda messy and not feeling very complete, also no tests written at all. For now, I wanted to limit the scope my changes to just the translation of the objc class to swift, and leave all the refactoring to the point when we start tackling and improving upload related code.

I may a TODO to remind us considering make this failable. How does this sound to you?

Reply:
Sounds good to me